src/unix/timer.c

```diff
@@ -94,9 +145,8 @@ int uv_timer_stop(uv_timer_t* handle) {
   if (!uv__is_active(handle))
     return 0;
 
-  heap_remove((struct heap*) &handle->loop->timer_heap,
-              (struct heap_node*) &handle->heap_node,
-              timer_less_than);
+  --handle->loop->timer_counter;
```

Comment:
I think `handle->loop->timer_counter--` would be cleaner (put the decrement operator at the end).